Add note to docs explaining the use of `"validation"` vs `"parse"` (#8080)
pydantic/pydantic: docs/concepts/models.md

differences that streamline certain workflows related to validation, serialization, and JSON schema generation.
You can find more discussion of this in the [Dataclasses](dataclasses.md) section of the docs.

> Untrusted data can be passed to a model and, after validation, Pydantic guarantees that the fields

Reviewer: i think this was fine since we do do parsing - e.g. of JSON.

of the resultant model instance will conform to the field types defined on the model.

> !!! note

Reviewer: ```suggestion
!!! note "Validation - deliberate misnomer"
```
or something?

>     The use of the terms "parse" and "validate" within Pydantic's context has been a source of confusion.

Reviewer: I don't think we need the big about "parse" and "validate", instead we should just explain why we use the word "validate".

    While these terms were previously used interchangeably, moving forward, we aim to exclusively employ "validate",
    with "parse" reserved specifically for discussions related to [JSON parsing](../concepts/json.md).

>     The potential confusion arises from the fact that, strictly speaking, Pydantic's primary focus

Reviewer: I think we want to say earlier on that "validation" is "the most widely understood colloquial term for the task pydantic performs" or similar.

Generally we should sound more like "this is the state of the world, we're not fighting it" rather than "we made a mistake long ago and we're sticking to that mistake.

Reviewer: You could say something like "for brevity we use the term "validation" to refer to the process of optional coercion and validation which Pydantic is known for".?

Untrusted data can be passed to a model and, after parsing and validation, Pydantic guarantees that the fields
of the resultant model instance will conform to the field types defined on the model.

> !!! note "Validation - a _deliberate_ misnomer"

Reviewer 2: ```suggestion
!!! note "Validation — a _deliberate_ misnomer"
```

Reviewer 2: Em dash

    ### TL;DR

    We use the term "validation" to refer to the process of constructing a model that adheres to specified types and

>     constraints. This task, which Pydantic is well known for, is most widely recognized as "validation" in colloquial terms.

Reviewer 2: ```suggestion
    constraints. This task, which Pydantic is well known for, is most widely recognized as "validation" in colloquial terms,
    even though in other contexts the term "validation" may be more restrictive.
```
can take it or leave it

Untrusted data can be passed to a model and, after parsing and validation, Pydantic guarantees that the fields
of the resultant model instance will conform to the field types defined on the model.

!!! note "Validation — a _deliberate_ misnomer"
    ### TL;DR

    We use the term "validation" to refer to the process of constructing a model that adheres to specified types and
    constraints. This task, which Pydantic is well known for, is most widely recognized as "validation" in colloquial terms,
    even though in other contexts the term "validation" may be more restrictive.

    ---

    ### The long version

    The potential confusion around the term "validation" arises from the fact that, strictly speaking, Pydantic's
    primary focus doesn't align precisely with the dictionary definition of "validation":

    > ### validation
    > _noun_
    > the action of checking or proving the validity or accuracy of something.

>     In Pydantic, the term "validation" refers to the process of constructing a model that adheres to specified

Reviewer: ```suggestion
    In Pydantic, the term "validation" refers to the process of instantiate a model (or other type) that adheres to specified
```

Author: Good call. Changed to "instantiating a model (or other type)" both here and in the TL;DR section.
